src/analyze/analyze.c

```diff
@@ -243,27 +243,28 @@ static int process_aliases(char *argv[], char *tempdir, char ***ret) {
         assert(ret);
 
         STRV_FOREACH(filename, strv_skip(argv, 1)) {
-                _cleanup_free_ char *src = NULL, *dst = NULL, *arg = NULL;
+                _cleanup_free_ char *src = NULL, *dst = NULL, *base = NULL;
                 char *parse_arg;
 
-                arg = strdup(*filename);
-                if (!arg)
-                        return -ENOMEM;
-
-                parse_arg = arg;
-                r = extract_first_word((const char **) &parse_arg, &src, ":", 0);
+                parse_arg = *filename;
+                r = extract_first_word((const char **) &parse_arg, &src, ":",
```

Comment:
no need to cast parse_arg like this anymore. You can declare the variable as const right-away. given the we don#t use basename() anymore we don't need it writable